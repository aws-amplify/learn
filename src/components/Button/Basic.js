import {css} from '@emotion/core';
import {useMemo} from 'react';
import {map} from 'ramda';
import Link from '../Link';
import {classNames} from '~/utilities';

const styles = css`
  display: flex;
  cursor: pointer;
  appearance: none;
  text-align: center;
  align-items: center;
  justify-content: center;

  &:active {
    outline: none;
  }
`;

const paddings = map(
  p =>
    css`
      padding: ${p};
    `,
  {
    small: `.3125rem`,
    medium: `.5625rem .75rem .5rem`,
    large: `.9375rem`,
  },
);

export default ({to, href, padding, className, ...rest}) => {
  const [Tag, uniqueProps] = useMemo(
    () => (to || href ? [Link, {to, href}] : ['button', {type: 'button'}]),
    [to, href],
  );

  return (
    <Tag
      css={[styles, paddings[padding || 'medium']]}
      className={classNames('button', className)}
      {...uniqueProps}
      {...rest}
    />
  );
};
