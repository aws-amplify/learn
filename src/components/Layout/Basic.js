import {css} from '@emotion/core';
import {mq, MAX_WIDTH} from '~/constants';
import {ToastContainer} from 'react-toastify';
import GlobalStyles from '../GlobalStyles';
import Footer from '../Footer';
import DiscordButton from "../DiscordButton";

export const style = css`
  display: flex;
  flex-direction: column;
  flex: 1;

  > .nav {
    z-index: 1;
  }

  > .body {
    display: flex;
    flex: 1;
    z-index: 0;

    > div {
      display: flex;
      flex-direction: row;
      flex: 1;
      margin: 0 auto;
      max-width: ${MAX_WIDTH};

      .menu {
        &.side {
          display: none;
          flex-shrink: 1;
          margin-top: 1.5625rem;

          ${mq.tablet} {
            display: flex;
          }
        }
      }

      .main {
        display: flex;
        flex-direction: column;
        flex: 1;
        z-index: 0;
        padding: 1rem 0;
      }
    }
  }
`;

export default ({header, main}) => (
  <>
    <GlobalStyles />
    <div css={style}>
      {header}
      <div className='body'>
        <div>
          <div className='main'>{main}</div>
        </div>
      </div>
    </div>
    <Footer />
    <ToastContainer />
    <DiscordButton />
  </>
);
