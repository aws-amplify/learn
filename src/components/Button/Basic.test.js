/* eslint no-undef: 0 */
import renderer from 'react-test-renderer';
import Basic from './Basic';

test('renders properly', () => {
  const component = renderer.create(<Basic>Hello</Basic>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
