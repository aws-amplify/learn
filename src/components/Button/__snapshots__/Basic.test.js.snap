// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<button
  className="button actionable three-dee rounded"
  css={
    Array [
      Object {
        "map": undefined,
        "name": "bxarfa",
        "next": undefined,
        "styles": "
  display: flex;
  cursor: pointer;
  appearance: none;
  text-align: center;
  align-items: center;
  justify-content: center;

  &:active {
    outline: none;
  }
",
      },
      Object {
        "map": undefined,
        "name": "jqbzm3",
        "next": undefined,
        "styles": "
      padding: .5625rem .75rem .5rem;
    ",
      },
    ]
  }
  onClick={[Function]}
  type="button"
>
  Hello
</button>
`;
